app/app.rb

```diff
@@ -293,14 +294,27 @@ def process_form_data(data, git)
 					file.relative_path.to_s
 				elsif data.keys.include?('___checkbox_unchecked')
 					data.keys.include?('___checkbox_checked')
+				elsif data.keys.include?("___datetime_date")
+					datetime = data.values.join(" ")
+					Time.parse(datetime) if datetime.present?
 				elsif data.keys.all? { |k| k =~ /\A\d+\Z/ }
 					data.to_a.sort_by {|x| x.first.to_i }.map(&:last).map { |v| process_form_data(v, git) }
 				else
-					data.merge(data) { |k,v| process_form_data(v, git) }
+					data.merge(data) do |k,v|
+						v = Date.parse(v) if is_date?(k, v)
+						process_form_data(v, git)
+					end
 				end
 			else
 				data
 			end
 		end
+
+		def is_date?(field_name, value)
+			(field_name.end_with?("_date") || field_name == "date") &&
+				!value.is_a?(Time) &&
```

Comment:
You check it's not a time and is a string. Seems redundant. Also, shouldn't being a date be allowed? Or I this is only used in process_form it's always a string...